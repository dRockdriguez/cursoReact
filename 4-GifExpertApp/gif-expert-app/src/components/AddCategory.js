import React, { useState } from "react";
import PropTypes  from "prop-types";

const AddCategory = ( { setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const onChangeHandler = ( { target } ) => {
    setInputValue(target.value);
  }

  const submitHandler = (e) => {
    e.preventDefault();
    
    if(inputValue.trim().length > 2) {
      setCategories(categories => [ inputValue, ...categories ])
      setInputValue("")
    }
  }

  return (
    <form onSubmit={ submitHandler }>
      <input 
        type="text"
        value={ inputValue }
        onChange={ onChangeHandler }/>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;
