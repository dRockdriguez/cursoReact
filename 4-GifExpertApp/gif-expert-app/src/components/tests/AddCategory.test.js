import React from 'react';
import AddCategory from '../AddCategory';
import { shallow } from 'enzyme';

describe('AddCategory', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory  setCategories={ setCategories }/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory  setCategories={ setCategories }/>);
    });

    test('Should render the component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Should not post value when submit', () => {
       const form = wrapper.find('form');

       const e = {
           preventDefault: () => {}
       }
       form.simulate('submit', e);

       expect(setCategories).not.toHaveBeenCalled();
    })

    test('Should post value when submit', () => {
        const eInput = {
            target: {
                value: 'Hola!'
            }
        }
        const eForm = {
            preventDefault: () => {}
        }

        wrapper.find('input').simulate('change', eInput);
        wrapper.find('form').simulate('submit', eForm);
 
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
     })
});