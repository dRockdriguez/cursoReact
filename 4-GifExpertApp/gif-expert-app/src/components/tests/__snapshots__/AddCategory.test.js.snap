// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Should render the component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
